openedx_learning/apps/authoring/publishing/models/draft_log.py

```diff
@@ -0,0 +1,307 @@
+
+    Scenario 7: old_version == new_version
+
+      This means that the data associated with the Draft version of an entity
+      has changed purely as a side-effect of some other entity changing.
+
+      The main example we have of this are containers. Imagine that we have a
+      Unit that is at v1, and has unpinned references to various Components that
+      are its children. The Unit's version does not get incremented when the
+      Components are edited, because the Unit container is defined to always get
+      the most recent version of those Components. We would only make a new
+      version of the Unit if we changed the metadata of the Unit itself (e.g.
+      the title), or if we added, removed, or reordered the children.
+
+      Yet updating a Component intuitively changes what we think of as the
+      content of the Unit. Users who are working on Units also expect that a
+      change to a Component will be reflected when looking at a Unit's
+      "last updated" info. The old_version == new_version convention lets us
+      represent that in a useful way because that Unit *is* a part of the change
+      set represented by a DraftChangeLog, even if its own versioned data hasn't
+      changed.
+    """
+    draft_change_log = models.ForeignKey(
+        DraftChangeLog,
+        on_delete=models.CASCADE,
+        related_name="records",
+    )
+    entity = models.ForeignKey(PublishableEntity, on_delete=models.RESTRICT)
+    old_version = models.ForeignKey(
+        PublishableEntityVersion,
+        on_delete=models.RESTRICT,
+        null=True,
+        blank=True,
+        related_name="+",
+    )
+    new_version = models.ForeignKey(
+        PublishableEntityVersion, on_delete=models.RESTRICT, null=True, blank=True
+    )
+
+    class Meta:
+        constraints = [
+            # A PublishableEntity can have only one DraftLogRecord per DraftLog.
+            # You can't simultaneously change the same thing in two different
+            # ways, e.g. set the Draft to version 1 and version 2 at the same
+            # time; or delete a Draft and set it to version 2 at the same time.
+            models.UniqueConstraint(
+                fields=[
+                    "draft_change_log",
+                    "entity",
+                ],
+                name="oel_dlr_uniq_dcl",
+            )
+        ]
+        indexes = [
+            # Entity (reverse) DraftLog Index:
+            #   * Find the history of draft changes for a given entity, starting
+            #     with the most recent (since IDs are ascending ints).
+            models.Index(
+                fields=["entity", "-draft_change_log"],
+                name="oel_dlr_idx_entity_rdcl",
+            ),
+        ]
+        verbose_name = _("Draft Change Log Record")
+        verbose_name_plural = _("Draft Change Log Records")
+
+
+class DraftSideEffect(models.Model):
+    """
+    Model to track when a change in one Draft affects other Drafts.
```

Comment:
This phrasing is confusing to me (is it accurate? because neither `cause` nor `effect` have any relationship to the `Draft` model that I can see). Could we say something like "Model to track when a draft change to an entity implicitly affects other entities such as parent containers" ?